bundles/org.openhab.binding.openweathermap/src/main/java/org/openhab/binding/openweathermap/internal/handler/OpenWeatherMapWeatherAndForecastHandler.java

```diff
@@ -376,12 +385,12 @@ private void updateHourlyForecastChannel(ChannelUID channelUID, int count) {
                     state = getQuantityTypeState(forecastData.getClouds().getAll(), PERCENT);
                     break;
                 case CHANNEL_RAIN:
-                    state = getQuantityTypeState(
-                            forecastData.getRain() == null ? 0 : forecastData.getRain().getVolume(), MILLI(METRE));
+                    Rain rain = forecastData.getRain();
+                    state = getQuantityTypeState(rain == null ? 0 : rain.getVolume(), MILLI(METRE));
```

Comment:
This should be something like:
```suggestion
                    state = getQuantityTypeState(rain == null || rain.getVolume() == 0 ? 0 : rain.getVolume(), MILLI(METRE));
```

The expected state is `0` instead of `UNDEF`.

Reply:
Sry, spelling mistake:

```suggestion
                    state = getQuantityTypeState(rain == null || rain.getVolume() == null ? 0 : rain.getVolume(), MILLI(METRE));
```

Reply:
That is also not needed, the body of `getQuantityTypeState` resolves that:

```java

    protected State getQuantityTypeState(@Nullable Number value, Unit<?> unit) {
        return (value == null) ? UnDefType.UNDEF : new QuantityType<>(value, unit);
    }

Reply:
:facepalm: I got it. Sry.